orttraining/orttraining/python/ort_trainer.py

```diff
@@ -674,6 +678,11 @@ def state_dict(self):
         return torch_state
 
     def load_state_dict(self, state_dict, strict=False):
+        if not self.session:
```

Comment:
When a session doesnt exist, what is the correct workflow? create a training session and then load the state? Should we add a warning to that scenario?

Reply:
I think probably the session should always exist on ORTTrainer(...) initialization? In order to do that it looks like sample inputs also need to be passed at that time.